ReactNativeClient/lib/services/searchengine/filterParser.ts

```diff
@@ -0,0 +1,113 @@
+
+interface Term {
+	name: string
+	value: string
+	negated: boolean
+}
+
+const makeTerm = (name: string, value: string): Term => {
+	if (name.startsWith('-')) { return { name: name.slice(1), value: value, negated: true }; }
+	return { name: name, value: value, negated: false };
+};
+
+const getTerms = (query: string) : Term[] => {
+	const terms: Term[] = [];
+	let inQuote = false;
+	let currentCol = '_';
+	let currentTerm = '';
+	for (let i = 0; i < query.length; i++) {
+		const c = query[i];
+
+		if (c === '"') {
+			currentTerm += c; // keep the quotes
+			if (inQuote) {
+				terms.push(makeTerm(currentCol, currentTerm));
+				currentTerm = '';
+				inQuote = false;
+			} else {
+				inQuote = true;
+			}
+			continue;
+		}
+
+		if (c === ' ' && !inQuote) {
+			if (!currentTerm) continue;
+			terms.push(makeTerm(currentCol, currentTerm));
+			currentCol = '_';
+			currentTerm = '';
+			continue;
+		}
+
+		if (c === ':' && !inQuote) {
+			currentCol = currentTerm;
+			currentTerm = '';
+			continue;
+		}
+
+		currentTerm += c;
+	}
+	if (currentTerm) terms.push(makeTerm(currentCol, currentTerm));
+	return terms;
+};
+
+const parseQuery = (query: string): Term[] => {
+	const validFilters = new Set(['any', 'title', '-title', 'body', '-body', 'tag',
+		'-tag', 'notebook', 'created', '-created',  '-updated', 'updated', 'type',
+		'iscompleted', 'latitude', '-latitude', 'longitude', '-longitude',
+		'altitude', '-altitude', 'resource', '-resource']);
```

Comment:
Your `makeTerm()` function above strips off the "-", which I think is the right thing to do. But doesn't it mean that all those `-filter` names in your set will never occur?